### ISSUE TITLE
cargo build --manifest-path <long-path> doesn't work on Windows

### ISSUE THREAD

**Issue by stepancheg** [rust-lang/cargo#8965]
```
cargo build --manifest-path <very-long-path>
```

doesn't work on Windows: reports

```
error: manifest path `<very-long-path>` does not exist
```

Script which reproduces the problem: https://github.com/stepancheg/test-win-long-paths/blob/master/test.sh
Failed job on GitHub actions: https://github.com/stepancheg/test-win-long-paths/runs/1532597301

## Why it is important

rustc has certain problems with long paths on Windows, which I need to reproduce. I tried to reproduce them with cargo and encountered this issue.

In certain build systems and certain CI environments build paths may be quite long. It is constant pain on Windows.

**Comment by ehuss** [comment]
There are several issues open for this (such as #7986, which explains a little why this is very difficult), and recent PRs such as #8874, #8881, #8964.  

Last time I checked, the MSVC linker didn't support long paths, so even if Cargo supported them, it would likely still not work.
(Though that information may be outdated, it has been a long time since I've tried.)

**Comment by ehuss** [comment]
Closing as a duplicate of #7986.

**Comment by dylni** [comment]
When #6198 is fixed, you can use a verbatim path instead (`\\?\long\path`). There's a PR open to fix it: #8881
